tutorial/LFRic/background/LFRic_kernel.md

```diff
@@ -0,0 +1,241 @@
+  !-----------------------------------------------------------------------------
+  type, public, extends(kernel_type) :: setval_field_w0_kernel_type
+    private
+    type(arg_type), dimension(2) :: meta_args = (/ &
+         arg_type(GH_FIELD, GH_INC, W0),           &
+         arg_type(GH_REAL,  GH_READ)               &
+         /)
+    integer :: iterates_over = CELLS
+  contains
+    procedure, nopass :: code => setval_field_w0_code
+  end type setval_field_w0_kernel_type
+
+  !-----------------------------------------------------------------------------
+  ! Contained functions/subroutines
+  !-----------------------------------------------------------------------------
+  public setval_field_w0_code
+
+  contains
+
+  subroutine setval_field_w0_code()
+
+  end subroutine setval_field_w0_code
+
+end module setval_field_w0_kernel_mod
+```
+
+As can be seen from the code above, the *LFRic kernels are objects, defined
+as derived types* that inherit from the abstract base type `kernel_type`
+
+```fortran
+type, public, extends(kernel_type) :: setval_field_w0_kernel_type
+```
+
+and this base type must be accessible to a kernel through a `use` statement
+
+```fortran
+use kernel_mod,        only: kernel_type
+```
+
+Looking at the `setval_field_w0_kernel_type` above, there are two main
+parts of an LFRic kernel:
+* Metadata that tells PSyclone how to operate on a kernel;
+* Subroutine with the argument list, declarations and executable code.
+
+All of this is stored in a module. The naming convention for the module,
+the kernel type and the executable subroutine is as follows:
+
+* Module name: `<base_name>_kernel_mod`;
+* Kernel type name: `<base_name>_kernel_type`;
+* Subroutine name: `<base_name>_code`.
+
+Here the `<base_name>` is `setval_field_w0`.
+
+### Kernel metadata
+
+The PSyclone metadata are stored as the *private* data within the definition
+of an individual kernel type. They come in two main forms: derived types
+(e.g. `arg_type` in the above example) and `integer` arguments (single-valued
+such as `iterates_over` above or arrays). Here we briefly explain the contents
+of `arg_type` and `iterates_over` metadata. For more information please refer
+to the [*Metadata* section](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#metadata)
+of the LFRic (Dynamo 0.3) API [user documentation.](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html)
+
+The `arg_type` in this example describes the two (hence `dimension(2)`)
+arguments that this kernel operates on:
+* A field argument (`GH_FIELD`) with the degrees of freedom (DoFs) on `W0`
+  function space that is being updated in this kernel (`GH_INC` access for
+  updating fields on continuous function spaces);
+* A `real`-valued scalar (`GH_REAL`) whose value is read (`GH_READ`) to
+  update the field with.
+
+The `GH_` prefix comes from "GungHo", the name of the dynamical core used in
+the LFRic model. Please refer to the documentation for more information on
```

Comment:
With the Dynamo interface and "GH" prefixes this document really highlights how badly we need to properly rename everything. But you are documenting what is, not what should be.